src/NRules/NRules.RuleModel/Builders/AggregateBuilder.cs

```diff
@@ -11,7 +11,7 @@ public class AggregateBuilder : RuleElementBuilder, IBuilder<AggregateElement>
     {
         private string _name;
         private Type _resultType;
-        private readonly Dictionary<string, LambdaExpression> _expressions = new Dictionary<string, LambdaExpression>();
```

Comment:
The idea was to use primitives in builders and static factories. So, I was thinking to use `KVP<string, LambdaExpression>` here and in the static factories in Element.cs.
What do you think?

Reply:
I did consider both Tuple and KVP, I think they are good too. Only issue is the names of the properties, which is a minor problem.

If using KVP, there's a lot of generic noise as there's no equivalent of Tuple.Create, unless I make one somewhere for our cases I guess? I think I'll have to retain the .Find/.FindSingle on IEnumerable<KVP> extension methods anyway.

Advantage of KVP is that places that currently create dictionaries can possibly keep working without changes assuming the KVP generic types match.

Reply:
Yeah, the generics noise of KVP is annoying. But I think the fact that you can just continue passing a dictionary in, when you don't have duplicates, is helpful.

Reply:
Ok, agreed, i'll make all of these changes and add to this PR shortly. Using dictionaries where appropriate will cut down on the noise quite a bit.